src/lavinmq/http/controller/prometheus.cr

```diff
@@ -291,40 +291,51 @@ module LavinMQ
       private def gc_metrics(writer)
         gc_stats = @amqp_server.gc_stats
 
-        writer.write({name: "gc_heap_size", value: gc_stats.heap_size,
+        writer.write({name: "gc_heap_size_bytes", value: gc_stats.heap_size,
                       type: "gauge",
                       help: "Heap size in bytes (including the area unmapped to OS)"})
+
         writer.write({name: "gc_free_bytes", value: gc_stats.free_bytes,
                       type: "gauge",
                       help: "Total bytes contained in free and unmapped blocks"})
+
         writer.write({name: "gc_unmapped_bytes", value: gc_stats.unmapped_bytes,
                       type: "gauge",
                       help: "Amount of memory unmapped to OS"})
-        writer.write({name: "gc_bytes_since_gc", value: gc_stats.bytes_since_gc,
-                      type: "counter",
-                      help: "Number of bytes allocated since the recent GC"})
-        writer.write({name: "gc_bytes_before_gc", value: gc_stats.bytes_before_gc,
+
+        writer.write({name: "gc_since_recent_collection_allocated_bytes", value: gc_stats.bytes_since_gc,
                       type: "gauge",
+                      help: "Number of bytes allocated since the recent GC"})
+
+        writer.write({name: "gc_before_recent_collection_allocated_bytes_total", value: gc_stats.bytes_before_gc,
+                      type: "counter",
                       help: "Number of bytes allocated before the recent GC (value may wrap)"})
-        writer.write({name: "gc_non_gc_bytes", value: gc_stats.non_gc_bytes,
+
+        writer.write({name: "gc_non_candidate_bytes", value: gc_stats.non_gc_bytes,
                       type: "gauge",
                       help: "Number of bytes not considered candidates for GC"})
-        writer.write({name: "gc_no", value: gc_stats.gc_no,
+
+        writer.write({name: "gc_cycles_total", value: gc_stats.gc_no,
                       type: "counter",
                       help: "Garbage collection cycle number (value may wrap)"})
-        writer.write({name: "gc_markers_m1", value: gc_stats.markers_m1,
+
+        writer.write({name: "gc_marker_threads_total", value: gc_stats.markers_m1,
```

Comment:
```suggestion
        writer.write({name: "gc_marker_threads", value: gc_stats.markers_m1,
```

I think the unit is "threads" and it's not a counter so no need for total, or am I overthinking this? :)

Reply:
IMO "threads" is enough. And yeah, not counter. It's probably a static number in our case 🤷